### PR TITLE
RuboCop changes.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This should get us one step closer to being able to merge Homebrew's and Cask's RuboCop  config.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Looking good but let's hold off merging until after 1.0.0 (so probably this time tomorrow).

File: Library/Homebrew/extend/ARGV.rb [hunk 1]
```diff
@@ -216,7 +216,8 @@ def build_from_source?
   end
 
   def build_all_from_source?
-    !!ENV["HOMEBREW_BUILD_FROM_SOURCE"]
+    return true if ENV["HOMEBREW_BUILD_FROM_SOURCE"]
```
Comment (jawshooah) [inline Library/Homebrew/extend/ARGV.rb]: Why not just replace `!!value` with `!value.nil?` here and elsewhere?

Comment (reitermarkus) [inline Library/Homebrew/extend/ARGV.rb]: To make sure, `!value.nil?` will only work for `ENV` variables, since those can't be `false`, or can they?

Comment (MikeMcQuaid) [inline Library/Homebrew/extend/ARGV.rb]:
```
mikebook # brew irb                (update-between-tags) [~/Documents/Homebrew]
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
>> ENV["HOMEBREW_DEVELOPER"] = false
TypeError: no implicit conversion of false into String
	from (irb):1:in `[]='
	from (irb):1
	from /Users/mike/Documents/Homebrew/Library/Homebrew/brew.rb:99:in `<main>'
>> 
```

so 👍 to `.nil?` from me.

**Comment by MikeMcQuaid** [comment]
Happy to 🚢 this when conflicts are resolved. @reitermarkus after that it'd be cool to unify Cask's rubocop and style too.

**Comment by MikeMcQuaid** [comment]
Nice. Would love to see us get a unified style on cask and the rest of Homebrew/brew as a next step.

**Comment by reitermarkus** [comment]
Yep, will create another PR *before* that though, to clean up most of the remaining offenses in `rubocop_todo.yml`.

**Comment by MikeMcQuaid** [comment]
🆒 👍

**Comment by ilovezfs** [comment]
This changed the API on search_taps, which now dies when passed a string.

**Pull request merged**